pkg/kv/kvserver/raft_log_queue.go

```diff
@@ -354,22 +354,30 @@ const (
 )
 
 // No assumption should be made about the relationship between
-// RaftStatus.Commit, FirstIndex, LastIndex. This is because:
+// RaftStatus.Commit, CompIndex, LastIndex. This is because:
 //   - In some cases they are not updated or read atomically.
-//   - FirstIndex is a potentially future first index, after the pending
+//   - CompIndex is a potentially future compacted index, after the pending
 //     truncations have been applied. Currently, pending truncations are being
 //     proposed through raft, so one can be sure that these pending truncations
 //     do not refer to entries that are not already in the log. However, this
 //     situation may change in the future. In general, we should not make an
-//     assumption on what is in the local raft log based solely on FirstIndex,
-//     and should be based on whether [FirstIndex,LastIndex] is a non-empty
-//     interval.
+//     assumption on what is in the local raft log based solely on CompIndex,
+//     and should be based on whether CompIndex < LastIndex.
 type truncateDecisionInput struct {
-	RaftStatus            raft.Status
-	LogSize, MaxLogSize   int64
-	LogSizeTrusted        bool // false when LogSize might be off
-	FirstIndex, LastIndex kvpb.RaftIndex
-	PendingSnapshotIndex  kvpb.RaftIndex
+	RaftStatus           raft.Status
+	LogSize, MaxLogSize  int64
+	LogSizeTrusted       bool // false when LogSize might be off
+	CompIndex            kvpb.RaftIndex
+	LastIndex            kvpb.RaftIndex
+	PendingSnapshotIndex kvpb.RaftIndex
+}
+
+// FirstIndex removes the first index of the raft log. The entry at this index
```

Comment:
Did you mean "returns"? In case of a fix-up, the commit to amend into is [7807cfa20d025b329dd1829f1fddd9ab701ba292](https://github.com/cockroachdb/cockroach/commit/7807cfa20d025b329dd1829f1fddd9ab701ba292).

Reply:
Done.